lib/proof-purpose/ProofPurpose.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Linked Data Signatures/Proofs
+ *
+ * @author Christopher Lemmer Webber
+ *
+ * @license BSD 3-Clause License
+ * Copyright (c) 2018 Digital Bazaar, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * Neither the name of the Digital Bazaar, Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+'use strict';
+
+// Note that classes implementing ProofPurpose in general should operate
+// on fields in expanded form for any of their own properties.  This is
+// because the number of fields that are possible given custom ProofPurpose
+// implementations are much larger than those mapped in the standard
+// ld-sigs / ocap-ld context.
+module.exports = class ProofPurpose {
+  constructor(injector) {
+    this.injector = injector;
+    // Must be overridden by child methods
+    this.uri = null;
+  }
+
+  async verify(document, proof, proofPurposeOptions) {
```

Comment:
I suppose the params are listed in stub as documentation?
```suggestion
  async verify(*/document, proof, proofPurposeOptions*/) {
```

Reply:
See my comment on arity previously.  I think the desire to comment out arguments when we're specifying what the API will use seems very strange.  Is this to prevent the linter from complaining, or something?